core/src/main/java/edu/stanford/futuredata/macrobase/pipeline/BasicBatchPipeline.java

```diff
@@ -21,44 +23,54 @@
 public class BasicBatchPipeline implements Pipeline {
     Logger log = LoggerFactory.getLogger(Pipeline.class);
 
-    private String inputURI = null;
+    // All classifier-specific fields need to be retrieved from ``conf''
+    private final PipelineConfig conf;
 
-    private String classifierType = "percentile";
-    private String metric = null;
-    private double cutoff = 1.0;
-    private boolean pctileHigh = true;
-    private boolean pctileLow = true;
+    // PipelineConfig params applicable to all classifiers
+    private final String inputURI;
+    private final String classifierType;
+    private final String metric;
 
+    // PipelineConfig params applicable to all summarizers
     private String summarizerType = "apriori";
     private List<String> attributes = null;
     private double minSupport = 0.01;
     private double minRiskRatio = 5.0;
 
 
     public BasicBatchPipeline (PipelineConfig conf) {
+        this.conf = conf;
+        // these fields must be defined explicitly in the conf.yaml file
         inputURI = conf.get("inputURI");
-
         classifierType = conf.get("classifier");
         metric = conf.get("metric");
-        cutoff = conf.get("cutoff");
-        pctileHigh = conf.get("includeHi");
-        pctileLow = conf.get("includeLo");
 
         summarizerType = conf.get("summarizer");
         attributes = conf.get("attributes");
         minRiskRatio = conf.get("minRiskRatio");
         minSupport = conf.get("minSupport");
-
     }
 
     public Classifier getClassifier() throws MacrobaseException {
         switch (classifierType.toLowerCase()) {
             case "percentile": {
```

Comment:
agree with Enum suggestion—but I was thinking it was more applicable to the predicates, since there's 6 predicates that we currently support. How many classifiers do you envision we'll have?